### PR TITLE
fix(setup): use correct setuptools backend

### PR DESCRIPTION
**Pull request by henryiii**
The default is `build-backend = "setuptools.build_meta:__legacy__"`, which is not ideal. This field really shouldn't have been optional, but it was.

The legacy backend injects the local path - that has to be done by hand with the non-legacy backend, since packages are discouraged from importing in setup.py. If that makes a difference, that will also have to be added here.

- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

Oops, didn't format the commit message. Can reformat after tests run.

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
Doesn't seem to like this.  Having problems with versioneer.

File: setup.py [hunk 1]
```diff
@@ -12,14 +12,21 @@
 from distutils.command.clean import clean
 import fnmatch
 import os
-from os.path import abspath, join as pjoin, relpath, split
+from os.path import join as pjoin, relpath, split
```
Comment (henryiii) [inline setup.py:15]:
```suggestion
from os.path import join as pjoin, relpath
```

**Comment by bashtage** [comment]
Fixed a minor flake issue.  Thanks.

**Comment by bashtage** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|maintenance/0.13.x|[](https://github.com/statsmodels/statsmodels/pull/8111)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by henryiii** [comment]
Personally, I'd use `setuptools_scm`, local importing into setup.py is discouraged, that's why it's only part of the `__legacy__` backend. Versioneer also doesn't support overiding the version via an environment variable, even though it's been asked for for years, while setuptools_scm does. But it was easy to fix.

**Pull request merged**